sdk/shared/src/main/scala/org/ergoplatform/sdk/JsonCodecs.scala

```diff
@@ -146,8 +146,10 @@ trait JsonCodecs {
       powNonce <- cursor.downField("powNonce").as[Coll[Byte]]
       powDistance <- cursor.downField("powDistance").as[sigma.BigInt]
       votes <- cursor.downField("votes").as[Coll[Byte]]
-    } yield new CHeader(id, version, parentId, adProofsRoot, stateRoot, transactionsRoot, timestamp, nBits,
-      height, extensionRoot, SigmaDsl.decodePoint(minerPk), SigmaDsl.decodePoint(powOnetimePk), powNonce, powDistance, votes)
+      unparsedBytes <- cursor.downField("unparsedBytes").as[Option[Coll[Byte]]]
```

Comment:
If it is encoded as Coll, then it should be decoded as Call. Maybe it should be saved as Option as well.

Reply:
this field made optional to be compatible with previous versions of clients. Thus we provide them with the new field but do not require it.

Reply:
Then the roundtrip property will only be partial. Shouldn't the "we provide them" part be decoupled from encoding? In other words, if we want the field to become required, this can be enforced in the object.

Reply:
why should it be required now, when at least until v7 bytes will not be used? Why to break clients now?

Reply:
> why should it be required now

You said "we provide them with the new field but do not require it" maybe I misunderstood it. Anyway, I think each encoder should have proper roundtrip property and here it is violated.

Reply:
It is violated when (and only) unparsed bytes are not provided, yes, which is perfectly showing what is done in order to support existing clients. Considering that breaking changes may happen in years or not happen at all, there is no reason to break the clients.

Reply:
By "clients" you mean any external code that directly use this headerDecoder? Then ScalaDoc would be helpful.